modules/mongodb/mongodb_test.go

```diff
@@ -118,3 +117,44 @@ func ExampleRunContainer_withCredentials() {
 	// Output:
 	// mongodb://root:password
 }
+
+func ExampleRunContainer_withReplicaSet() {
+	ctx := context.Background()
+
+	container, err := mongodb.RunContainer(ctx,
+		testcontainers.WithImage("mongo:6"),
+		mongodb.WithUsername("root"),
+		mongodb.WithPassword("password"),
+		mongodb.WithReplicaSet("rs0"),
+		testcontainers.WithWaitStrategy(wait.ForLog("Waiting for connections")),
```

Comment:
IMO, the replicaSet opt-in should be just `WithReplicaSet()` and it should be stored in a flag used to set the right wait strategy and exec the commands in the container lifecycle. However, not sure how important is setting the replica set name but having a default could be just enough?